### PR TITLE
CLOUDSTACK-9569: add router.aggregation.command.each.timeout to agent…

### PR DESCRIPTION
**Pull request by abhinandanprateek**
….properties

### CODE REVIEW

**Comment by koushik-das** [comment]
@abhinandanprateek Looks like there is another PR #1856 related to this.

**Comment by borisstoyanov** [comment]
Testing has PASSED
There's a way to override the hardcoded timeout of 10 mins now by setting the "router.aggregation.command.each.timeout" in the "/etc/cloudstack/agent/agent.properties" file.
Testing steps:
Set the router.aggregation.command.each.timeout=1
Added a sleep of 15 seconds in the router_proxy.sh
Restarted the agent 
Started a new VM with a new network
In the log the following observations were noted:

```
2017-01-20 12:04:05,787 DEBUG [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-2:null) Executing: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.3.57 -c /var/cache/cloud/VR-a20603a7-8e10-4378-96bb-7a2dbc7c6c0b.cfg
2017-01-20 12:04:06,792 WARN  [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-2:null) Timed out: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.3.57 -c /var/cache/cloud/VR-a20603a7-8e10-4378-96bb-7a2dbc7c6c0b.cfg .  Output is:
2017-01-20 12:05:20,419 DEBUG [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-1:null) Executing: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.0.41 -c /var/cache/cloud/VR-090cae6a-f07d-40bb-9f19-809ccdcca16b.cfg
2017-01-20 12:05:21,423 WARN  [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-1:null) Timed out: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.0.41 -c /var/cache/cloud/VR-090cae6a-f07d-40bb-9f19-809ccdcca16b.cfg .  Output is:
2017-01-20 12:06:33,620 DEBUG [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-1:null) Executing: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.2.33 -c /var/cache/cloud/VR-63f52dbd-b710-4e57-a21b-1f4bcd146ec3.cfg
2017-01-20 12:06:34,624 WARN  [kvm.resource.LibvirtComputingResource] (agentRequest-Handler-1:null) Timed out: /usr/share/cloudstack-common/scripts/network/domr/router_proxy.sh vr_cfg.sh 169.254.2.33 -c /var/cache/cloud/VR-63f52dbd-b710-4e57-a21b-1f4bcd146ec3.cfg .  Output is:
```

VR Failed to start with a Message "Unable to start a VM due to insufficient capacity" 
I think we could add a log message saying that it timed out in the agent log, because now it does not leave any clue where the time out is.

**Comment by rhtyd** [comment]
LGTM. @abhinandanprateek is this PR different or is a subset of https://github.com/apache/cloudstack/pull/1856

**Comment by rhtyd** [comment]
VR_SCRIPT_EXEC_TIMEOUT is 10 minutes, LGTM checked.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@borisstoyanov is this good to merge or does Wei's fix in #1856 good enough?

**Comment by borisstoyanov** [comment]
I think they are both addressing the same thing although adding changes in different files. Maybe needs to be reviewed by another dev @rhtyd

**Comment by rhtyd** [comment]
@borisstoyanov given Wei's fix, and propagation, this may be unnecessary but @ustcweizhou can comment, what do you think Wei?

File: core/src/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java [hunk 1]
```diff
@@ -272,7 +272,10 @@ public boolean configure(final String name, final Map<String, Object> params) th
         _port = NumbersUtil.parseInt(value, 3922);
 
         value = (String)params.get("router.aggregation.command.each.timeout");
-        _eachTimeout = Duration.standardSeconds(NumbersUtil.parseInt(value, 10));
+        _eachTimeout = Duration.standardSeconds(NumbersUtil.parseInt(value, (int)VRScripts.VR_SCRIPT_EXEC_TIMEOUT.getStandardSeconds()));
```
Comment (rhtyd) [inline core/src/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java]: @borisstoyanov however this fix makes sense to use the default value when not parse-able.

**Pull request merged**